jdaviz/core/loaders/resolvers/resolver.py

```diff
@@ -0,0 +1,232 @@
+import warnings
+from traitlets import Bool, List, Unicode, observe
+
+from jdaviz.core.template_mixin import PluginTemplateMixin, SelectPluginComponent, with_spinner
+from jdaviz.core.registries import loader_parser_registry, loader_importer_registry
+from jdaviz.core.user_api import LoaderUserApi
+
+__all__ = ['BaseResolver']
+
+
+class FormatSelect(SelectPluginComponent):
+    def __init__(self, plugin, items, selected, default_mode='first'):
+        """
+        Parameters
+        ----------
+        plugin
+            the parent plugin object
+        items : str
+            the name of the items traitlet defined in ``plugin``
+        selected : str
+            the name of the selected traitlet defined in ``plugin``
+        default_mode : str, optional
+            What mode to use when making the default selection.  Valid options: first, default_text,
+            empty.
+        """
+        self._importers = {}
+        super().__init__(plugin,
+                         items=items,
+                         selected=selected,
+                         default_mode=default_mode)
+
+    def _is_valid_item(self, item):
+        return super()._is_valid_item(item, locals())
+
+    @observe('filters')
+    def _update_items(self, msg={}):
+        if not self.plugin.is_valid:
+            self.items = []
+            self._apply_default_selection()
+            return
+
+        # check for valid parser > importer combinations given the current filters
+        # and resolver inputs
+        try:
+            parser_input = self.plugin()
+        except Exception:
+            self.items = []
+            self._apply_default_selection()
+            return
+
+        all_resolvers = []
+        self._importers = {}
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
```

Comment:
any ideas why warnings while searching for matching loaders are still slipping through would be super helpful... otherwise we can probably defer that to a follow-up, but definitely something we need to get right before making this public 🤔

Reply:
If the warning is from a logger, then this catch_warnings won't catch it. Exactly what warning you saw?